dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -815,3 +876,129 @@ CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtensi
     THROW_IF_WIN32_ERROR(UuidFromStringW(textRpcString, &clsid));
     return clsid;
 }
+
+bool IsOptionEnabled(PCWSTR name)
+{
+    WCHAR value[1 + 1]{};
+    if (GetEnvironmentVariableW(name, value, ARRAYSIZE(value)) == 1)
+    {
+        if (*value == L'0')
+        {
+            return false;
+        }
+        else if (*value == L'1')
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+HRESULT MddBootstrapInitialize_Log(
+    HRESULT hrInitialize,
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion) noexcept try
```

Comment:
Am curious what's driving the need to write to event log. Seems like a lot of additional complexity for zero value.

Reply:
add an option to make it explicit and use delayload?

Reply:
Anyone can write to the EventLog. You need admin privilege to define a new channel or source (which we don't); if you `ReportEvent` to a non-existent source the event goes to the Application channel. No special powers needed.

This is unconditional as there's no persisted note if problems occur, and this should be a rare event and only logged if something went wrong. Having that persisted log event becomes critical to troubleshoot real world problems.

Notice we don't log anything on success. That's not problematic or helpful, and it _would_ be a ton of data for no value. Check out the Application channel on your local PC and you'll find a ton of events (especially Informational) for routine operations.

>Seems like a lot of additional complexity for little value

The complexity is to handle the "if something goes wrong make sure _something_ is logged." We have too much 1st hand experience troubleshooting problems (even just WinAppSDK issues the past 12 months) not to value this logging.

And it's not really a lot of complexity. More than a dumb sprintf statement but not really that much. Reliability and supportability often require more than simplstic sprintf or string+string 'golden path' code entails.

Reply:
> Nope. Anyone can write to the EventLog.

That may hold true in the commercial sector, but it isn't true in DoD/IC world. And because it's not universally true, that means my app's runtime suddenly writing to the Event Log has now introduced an additional dependency I need to worry about. In this case, developers can't even opt-out. Just my thinking, not posing this as a problem.

Reply:
>If it gracefully fails, then no problem. Just thinking out loud, thanks.

It does :-)

`MddBootstrapInitialize_Log()` is best effort so worst case no harm no foul